sources/packages/backend/libs/integrations/src/services/sfas/sims-to-sfas.service.ts

```diff
@@ -0,0 +1,161 @@
+    const [latestBridgeFileLog] = await this.sfasBridgeLogRepo.find({
+      select: { id: true, referenceDate: true },
+      order: { referenceDate: "DESC" },
+      take: 1,
+    });
+    return latestBridgeFileLog ? latestBridgeFileLog.referenceDate : null;
+  }
+
+  /**
+   * Log the details of bridge file that was sent to SFAS.
+   * @param referenceDate date when the bridge file data was extracted.
+   * @param fileName bridge file name.
+   */
+  async logBridgeFileDetails(
+    referenceDate: Date,
+    fileName: string,
+  ): Promise<void> {
+    await this.sfasBridgeLogRepo.insert({
+      referenceDate,
+      generatedFileName: fileName,
+    });
+  }
+
+  /**
+   * Get all student ids of students who have one or more updates
+   * since the date provided.
+   * The updates can be one or more of the following:
+   * - Student or User data
+   * - Sin validation data
+   * - Cas supplier data
+   * - Overawards data
+   * @param modifiedSince the date after which the student data was updated.
+   * If not provided, all the students with at least one submitted application will be returned.
+   */
+  async getAllStudentsWithUpdates(modifiedSince?: Date): Promise<number[]> {
+    const applicationStudentQuery = this.applicationRepo
+      .createQueryBuilder("application")
+      .select(["application.id", "student.id"])
+      .distinctOn(["student.id"])
+      .innerJoin("application.student", "student")
+      .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidation")
+      .innerJoin("student.casSupplier", "casSupplier")
+      .leftJoin("student.overawards", "overaward")
+      .where("application.applicationStatus != :overwritten", {
+        overwritten: ApplicationStatus.Overwritten,
+      })
+      .andWhere("application.currentAssessment is not null");
+    if (modifiedSince) {
+      applicationStudentQuery
+        .andWhere(
+          new Brackets((qb) => {
+            qb.where("student.updatedAt > :modifiedSince")
+              .orWhere("user.updatedAt > :modifiedSince")
+              .orWhere("sinValidation.updatedAt > :modifiedSince")
+              .orWhere("casSupplier.updatedAt > :modifiedSince")
+              .orWhere("overaward.updatedAt > :modifiedSince");
+          }),
+        )
+        .setParameter("modifiedSince", modifiedSince);
+    }
+    const applicationsWithStudentUpdates =
+      await applicationStudentQuery.getMany();
+    // Extract the student ids from the applications.
+    const modifiedStudentIds = applicationsWithStudentUpdates.map(
+      (application) => application.student.id,
+    );
+    return modifiedStudentIds;
+  }
+  /**
+   * Get student details of students who have one or more updates.
+   * @param studentIds student ids.
+   * @returns student details.
+   */
+  async getStudentRecordsByStudentIds(
+    studentIds: number[],
+  ): Promise<StudentDetail[]> {
+    const queryResult = await this.studentRepo
```

Comment:
For clarification, the group by approached was chosen because the MSFAA was removed and the student data is not supposed to grow, hence it looks better and simpler, right?

Reply:
Yes that is right.